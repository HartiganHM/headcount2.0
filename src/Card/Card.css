.card {
  align-items: center;
  border: 1px solid gray;
  display: flex;
  flex: 1;
  flex-direction: column;
  height: 16rem;
  padding: 1rem;
  margin: 1rem;
  text-align: center;
  max-width: 12rem;
  min-width: 12rem; }
  .card .district-name {
    font-size: 1.2rem; }
  .card .district-data {
    font-size: 0.8rem;
    text-align: left; }
    .card .district-data .greaterThan {
      color: blue; }
    .card .district-data .lessThan {
      color: red; }

.selected {
  background-color: rgba(0, 197, 0, 0.37); }
