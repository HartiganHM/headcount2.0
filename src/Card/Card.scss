.card {
    align-items: center;
    border: 1px solid gray;
    display: flex;
    flex: 1;
    flex-direction: column;
    height: 16rem;
    padding: 1rem;
    margin: 1rem;
    text-align: center;
    max-width: 12rem;
    min-width: 12rem;

    .district-name {
        font-size: 1.2rem;
    }

    .district-data {
        font-size: 0.8rem;
        text-align: left;

        .greaterThan {
            color: blue;
        }

        .lessThan {
            color: red;
        }
    }
}
