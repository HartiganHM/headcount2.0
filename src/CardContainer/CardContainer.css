.card-container {
  justify-content: space-between;
  display: flex;
  flex-wrap: wrap;
  padding: 0 1.2rem 1.2rem; }
  .card-container:after {
    content: "";
    flex: auto; }
