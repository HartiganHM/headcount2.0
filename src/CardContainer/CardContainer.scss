.card-container {
    justify-content: space-around;
    display: flex;
    flex-wrap: wrap;
    padding: 0 1.2rem;

    &:after {
        content: "";
        flex: auto;
    }
}