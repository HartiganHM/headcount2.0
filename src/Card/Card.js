import React from 'react';
import PropTypes from 'prop-types';
import './Card.css';

const Card = ({ district, data }) => {
  const districtData = Object.keys(data).map(year => {
    return (
      <li key={year}>
        {year}: {data[year]}
      </li>
    );
  });

  return (
    <div className='card'>
      <span className='district-name'>{district}</span>
      <ul className='district-data'>{districtData}</ul>
    </div>
  );
};

Card.propTypes = {
  district: PropTypes.string,
  data: PropTypes.object
};

export default Card;
