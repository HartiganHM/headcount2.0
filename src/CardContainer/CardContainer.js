import React from 'react';
import PropTypes from 'prop-types';
import Card from '../Card/Card';
import './CardContainer.css';

const CardContainer = ({ data, selectCard, removeCard, selectedArray }) => {
  if (data) {
    let type;
    let clickFunction;

    const allCards = Object.keys(data).map(district => {
      const dist = data[district];
      let dist1 = selectedArray.length ? selectedArray[0].location : null;
      let dist2 = selectedArray.length > 1 ? selectedArray[1].location : null;

      if (selectedArray.length && dist.location === dist1) {
        type = 'card selected';
        clickFunction = removeCard;
      } else if (selectedArray.length && dist.location === dist2) {
        type = 'card selected';
        clickFunction = removeCard;
      } else {
        type = 'card';
        clickFunction = selectCard;
      }

      return (
        <Card district={dist.location} data={dist.data} key={dist.location} clickFunction={clickFunction} type={type}/>
      );
    });

    return <div className="card-container">{allCards}</div>;
  }
  return null;
};

CardContainer.propTypes = {
  data: PropTypes.object
};

export default CardContainer;