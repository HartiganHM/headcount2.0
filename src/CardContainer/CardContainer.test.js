import React from 'react';
import CardContainer from './CardContainer';
import Card from '../Card/Card';
import { shallow, mount } from 'enzyme';
import DistrictRepository from '../helper';
import importedData from '../../data/kindergartners_in_full_day_program.js';

describe('Card Container Tests', () => {
  let renderedCardContainer;
  let mockData;
  let mockFunc;
  let mockArray;

  beforeEach(() => {
    mockData = {
      colorado: { location: 'colorado', data: { 2005: 0.8 } },
      ADAMSCOUNTY: { location: 'ADAMS COUNTY', data: { 2005: 0.6 } }
    };
    mockFunc = jest.fn();
    mockArray = [
      { location: 'yuma', data: { 2005: 0.8 } },
      { location: 'denver', data: { 2005: 0.6 } }
    ];
    renderedCardContainer = shallow(
      <CardContainer
        data={mockData}
        selectCard={mockFunc}
        removeCard={mockFunc}
        selectedArray={mockArray}
      />
    );
  });

  it('CardContainer should exist', () => {
    expect(renderedCardContainer).toBeDefined();
  });

  it('CardContainer should match the snapshot', () => {
    expect(renderedCardContainer).toMatchSnapshot();
  });

  it('CardContainer should render Cards', () => {
    // renderedCardContainer = mount(<CardContainer
    //   data={mockData}
    //   selectCard={mockFunc}
    //   removeCard={mockFunc}
    //   selectedArray={mockArray}
    // />)
    console.log(renderedCardContainer.debug());
    console.log(renderedCardContainer.find('Card').length);
    // expect(renderedCardContainer.find('Card').length).toEqual(2);

    mockData = new DistrictRepository(importedData);
    renderedCardContainer = shallow(<CardContainer data={mockData} />);

    const expectedLength = 181;

    console.log(renderedCardContainer.debug());

    // expect(renderedCardContainer.find('Card').length).toEqual(expectedLength);
  });
});
