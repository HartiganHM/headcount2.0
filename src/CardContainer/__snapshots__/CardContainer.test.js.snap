exports[`Card Container Tests CardContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    data={
        Object {
            "ADAMSCOUNTY": Object {
              "data": Object {
                "2005": 0.6,
              },
              "location": "ADAMS COUNTY",
            },
            "colorado": Object {
              "data": Object {
                "2005": 0.8,
              },
              "location": "colorado",
            },
          }
    }
    removeCard={[Function]}
    selectCard={[Function]}
    selectedArray={Array []} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          clickFunction={[Function]}
          data={
                    Object {
                              "2005": 0.8,
                            }
          }
          district="colorado"
          type="card" />,
        <Card
          clickFunction={[Function]}
          data={
                    Object {
                              "2005": 0.6,
                            }
          }
          district="ADAMS COUNTY"
          type="card" />,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "colorado",
        "nodeType": "function",
        "props": Object {
          "clickFunction": [Function],
          "data": Object {
            "2005": 0.8,
          },
          "district": "colorado",
          "type": "card",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "ADAMS COUNTY",
        "nodeType": "function",
        "props": Object {
          "clickFunction": [Function],
          "data": Object {
            "2005": 0.6,
          },
          "district": "ADAMS COUNTY",
          "type": "card",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            clickFunction={[Function]}
            data={
                        Object {
                                    "2005": 0.8,
                                  }
            }
            district="colorado"
            type="card" />,
          <Card
            clickFunction={[Function]}
            data={
                        Object {
                                    "2005": 0.6,
                                  }
            }
            district="ADAMS COUNTY"
            type="card" />,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "colorado",
          "nodeType": "function",
          "props": Object {
            "clickFunction": [Function],
            "data": Object {
              "2005": 0.8,
            },
            "district": "colorado",
            "type": "card",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "ADAMS COUNTY",
          "nodeType": "function",
          "props": Object {
            "clickFunction": [Function],
            "data": Object {
              "2005": 0.6,
            },
            "district": "ADAMS COUNTY",
            "type": "card",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
